third_party/grpc/bazel.patch

```diff
@@ -0,0 +1,288 @@
+From 6eb4a396945bbe89831ce1d205106c1348f376e3 Mon Sep 17 00:00:00 2001
+From: Mark Schaller <mschaller@google.com>
+Date: Thu, 9 Jan 2020 19:50:57 -0500
+Subject: [PATCH] Adapt gRPC bazel files for use in Bazel
+
+Simplifies cc_grpc_library. Removes well_known_proto-related
```

Comment:
The only change I would suggest is in the diff's commit message, which imho should be more detailed to make manual rebasing easier for future maintainers.

Something like this:

Simplifies cc_grpc_library:
* removes grpc_only arg (forced to True);
* removes proto_only, well_known_protos (forced to False);
* adds extra_cc_library_kwargs

Removes well_known_proto-related functionality throughout.
Renames generate_cc's _protoc attr to protoc. Relativizes
load statements. Adds a filegroup, "srcs", to the package.

Reply:
Thanks, good idea! I've changed the 2nd-to-last paragraph of the commit message to this:

```
This includes a patch file, bazel.patch, which adapts gRPC's BUILD/bzl
sources for use by Bazel.

This simplifies the cc_grpc_library macro and its deps:
* removes cc_grpc_library's grpc_only parameter (now always true)
* removes its proto_only, well_known_protos parameters (now always
    false)
* removes its use_external parameter (it was previously unused)
* adds to it an extra_cc_library_kwargs parameter (forwarded to its
    generated cc_library rule)
* removes "well_known_proto" functionality from generate_cc.bzl and
    protobuf.bzl
* changes generate_cc's _protoc attribute to protoc
* sets cc_grpc_library's generate_cc call's plugin and protoc arguments
    to Bazel's targets ("//third_party/grpc:cpp_plugin" and
    "//third_party/protobuf:protoc", respectively)
* relativizes load statements throughout
* adds "srcs" filegroup to the package's BUILD file and includes it in
    //third_party/grpc:srcs

```

Reply:
I'm sorry for the unclear wording. I meant adding more detail to the message *in the new `bazel.patch` file* - because that's what someone will be looking at 1 year from now when they run `patch -p3 < bazel.patch` and get conflicts.

Reply:
Oh, I see! No prob--done.